src/arena.rs

```diff
@@ -37,6 +37,10 @@ impl<T> Arena<T> {
     pub fn new() -> Arena<T> {
         Self::default()
     }
+    
+    pub fn get_node_id(node: Node<T>) -> Option<NodeId> where T: PartialEq {
```

Comment:
Can you please add a doc comment and example/test for that?

Reply:
Yes. Sorry, I should have raised an issue and all – was more experimenting to see if this would work.

Reply:
Yes take your time, no worries at all. :)